### PR TITLE
feat: add Automatic-Module-Name to manifest (#324)

### LINKED ISSUE #324: Add Automatic-Module-Name to manifest
https://github.com/GoogleCloudPlatform/cloud-opensource-java/blob/master/library-best-practices/JLBP-20.md

### PR DESCRIPTION
**Pull request by man0s**
2/3 for #324 
As stated in #325 by @elharo , Automatic-Module-Name need to be in submodules and not in parent.
lmk if everything is okay.

### CODE REVIEW

File: bom/pom.xml [hunk 1]
```diff
@@ -96,6 +96,17 @@
           <skip>true</skip>
         </configuration>
       </plugin>
+      <plugin>
```
Comment (chingor13) [inline bom/pom.xml]: The bom doesn't provide any sources, so this configuration is unnecessary.

File: appengine/pom.xml [hunk 2]
```diff
@@ -35,6 +35,17 @@
         <groupId>org.apache.maven.plugins</groupId>
         <artifactId>maven-javadoc-plugin</artifactId>
       </plugin>
+      <plugin>
+      	<groupId>org.apache.maven.plugins</groupId>
+        <artifactId>maven-jar-plugin</artifactId>
+	      <configuration>
+	        <archive>  
+	          <manifestEntries>
+	            <Automatic-Module-Name>com.google.auth</Automatic-Module-Name>
```
Comment (chingor13) [inline appengine/pom.xml]: This module name should be `com.google.auth.appengine`

File: oauth2_http/pom.xml [hunk 3]
```diff
@@ -47,6 +47,17 @@
           <ignoredUnusedDeclaredDependencies>com.google.auto.value:auto-value</ignoredUnusedDeclaredDependencies>
         </configuration>
       </plugin>
+      <plugin>
+        <groupId>org.apache.maven.plugins</groupId>
+        <artifactId>maven-jar-plugin</artifactId>
+        <configuration>
+          <archive>  
+            <manifestEntries>
+              <Automatic-Module-Name>com.google.auth</Automatic-Module-Name>
```
Comment (chingor13) [inline oauth2_http/pom.xml]: Please change this one to `com.google.auth.oauth2`

File: appengine/pom.xml [hunk 4]
```diff
@@ -35,6 +35,17 @@
         <groupId>org.apache.maven.plugins</groupId>
         <artifactId>maven-javadoc-plugin</artifactId>
       </plugin>
+      <plugin>
+      	<groupId>org.apache.maven.plugins</groupId>
+        <artifactId>maven-jar-plugin</artifactId>
+	      <configuration>
```
Comment (chingor13) [inline appengine/pom.xml]: Please fix the indentation here to 2 spaces.

Comment (man0s) [reply appengine/pom.xml]: Thanks, got it!

**Comment by man0s** [comment]
Thanks @elharo & @chingor13 for the suggestions.

**Comment by chingor13** [comment]
The credentials module was included. It's the `com.google.auth` module as that artifact is providing classes in that root package.

**Comment by elharo** [comment]
My mistake. Sorry.

**Pull request merged**